### ISSUE TITLE
Is a .Net wrapper possible?

### ISSUE THREAD

**Issue by BrianLima** [opticalgenesis/AuraCUE#1]
Hi @opticalgenesis, i'm developing some apps and think that your AuraCUE Project would be really usefull, i have some RGB RAM from TeamGROUP that are Asus Aura compatible but i have two problems: 

- I can't compile your Project on my PC
![image](https://user-images.githubusercontent.com/1337383/29314949-d4323c34-8196-11e7-879c-e497d1d11429.png) because it seems that many paths on your Project are hardcoded inside it.

- I can't add it as reference inside a .Net Project, already tried interop calling it, converting with TlbImp.exe and no luck with either.

Unfortunatelly i'm not very familiar with C/C++ so i won't venture inside your code to try to fix something and end-up breaking everything.

Thanks in advance, and keep up the good work!

**Comment by BrianLima** [comment]
Update: If I create such directory on my PC and place the project on it, I can compile, but the teste project won't do anything, It won't recognize my RGB RAM as a AURA device. QueryDeviceNames() return no devices on my system.

**Comment by opticalgenesis** [comment]
Hi thanks for these comments.

I'll dive in and sort out the path dependencies ASAP.

As for QueryDeviceNames() returning null, do you have the ASUS Aura program installed on your machine?

And finally, R.E the .NET wrapper, I see no reason why it wouldn't be possible. I'll look into it once I've sorted out issues on the C++ side.

**Comment by BrianLima** [comment]
I do have the Asus Aura software installed, BUT i have an MSI Tomahawk B350 mobo, the software can read the RAM with no problem as you can see:

![image](https://user-images.githubusercontent.com/1337383/29317542-c1a560ee-81a2-11e7-85bf-d9a4610b4842.png)

It's identified in the same way that the Trident Z RGB RAM is, Asus aura doesn't provide any more details, but i can control each LED on both RAM sticks individually. Is there a way that i can find the id that CLSIDFromProgID looks for?

Also, thanks for considering the .Net wrapper!

**Comment by opticalgenesis** [comment]
Hm, interesting.

I went looking for documentation (it's been a while since I've worked on this you see) and it turns out there's a whole new SDK release with a completely different structure.

I am about to start integrating the changes. Once this task is done we can come back and see if this is an issue with my code, or the old SDK release.

**Comment by BrianLima** [comment]
That's cool, i'll be watching for future updates, when you close #2 i'll try testing again.

**Comment by opticalgenesis** [comment]
I've discovered the root of the problem you reported, in the build of the Aura SDK used in the version published here, an Aura enabled motherboard is required for the SDK to work.

I'm hoping this is fixed by ASUS pretty quickly, but as you can see by #3 and #4 there are some pretty significant bugs in the newer release.

I'll still work on integrating it, but I wouldn't have too high expectations that it'll fix your issue.

**Comment by BrianLima** [comment]
> in the build of the Aura SDK used in the version published here, an Aura enabled motherboard is required for the SDK to work.

This just remembers me how I hate idiot proprietary tech from the deep of my heart with all my forces.

I'll start posting on that forum so maybe we can get some progress, I hate to see it go this way.

**Comment by opticalgenesis** [comment]
Yeah, whilst it is kind of understandable in a beta release like this is; I would very much like to see more compatibility with 3rd party hardware sooner rather than later.

It is also possible that this is fixed in the current SDK release, but as there is next-to-no documentation I can find, I have no way of knowing

**Comment by BrianLima** [comment]
I tried looking up the oficial SDK not long ago and I had this very problem, no documentation and the dev only told it "should be working", that's how I ended up on your project.

Hope those incompatibilities are solved soon.

**Comment by BennyAlex** [comment]
@opticalgenesis Hey, I was wondering what are your plans about this project? Do you still working on it? Will there a way to call AuraCUE from C#?

**Comment by opticalgenesis** [comment]
Work is on hold until tomorrow when I can run some local tests.

Once the C++ code is functional, I'll get to working on access from other languages

**Comment by BennyAlex** [comment]
I am an mainly a javascript programmer, do you think I can call the C++ from node.js? Your programm is just C++ or is there anything else? I know that in node.js it is possible to call C++ libaries...

**Comment by opticalgenesis** [comment]
I can't comment on the core CUE and Aura SDKs as I didn't write them, though they do seem to be pure C++; but as for my additions, they are also just C++.

Having never touched JS myself, I don't know how much success you'll have using it, but if you can call C++ from it then I see no reason why it wouldn't work
